.github/workflows/codecov.yml

```diff
@@ -0,0 +1,30 @@
+name: Test and coverage
+
+on: [push, pull_request]
+
+permissions:  # added using https://github.com/step-security/secure-workflows
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@4d991eb9b905ef189e4c376166672c3f2f230481
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
```

Comment:
any particular reason for using a (I assume) commit hash rather than just the latest `v4`? Same goes for other actions as well

Reply:
I think that's what dependabot did.